server/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -198,12 +201,33 @@ Metadata upgradeMetadataForNode(Metadata metadata,
     public static Metadata upgradeMetadata(Metadata metadata,
                                     MetadataIndexUpgradeService metadataIndexUpgradeService,
                                     MetadataUpgrader metadataUpgrader) {
-        // upgrade index meta data
         boolean changed = false;
         final Metadata.Builder upgradedMetadata = Metadata.builder(metadata);
+
+        // carries upgraded IndexTemplateMetadata to MetadataIndexUpgradeService.upgradeIndexMetadata
+        final Map<String, IndexTemplateMetadata> upgradedIndexTemplateMetadata = new HashMap<>();
+
+        // upgrade current templates
+        if (applyPluginUpgraders(
+            metadata.getTemplates(),
```

Comment:
Isn't it possible to utilise the existing upgrades without the changes here?

Reply:
I wanted to reference the upgrades done on template mappings while upgrading index mappings. Could you explain a little more details how?